vulcanai/datasets/tabulardataset.py

```diff
@@ -498,40 +548,55 @@ def split(self, split_ratio=0.7, stratified=False, strata_field='label'):
             stratified: Boolean
                 whether the sampling should be stratified.
                     Default is False.
-            strata_field: String
-                name of the examples Field stratified over.
-                Default is 'label' for the conventional label field.
+            stratum_column: String
+                name of the examples column stratified over.
+                Default is 'label_column'
 
         Returns:
             datasets: Tuple of TabularDatasets
                 Datasets for train, validation, and
                 test splits in that order, if the splits are provided.
 
         """
-        if stratified:
-            raise NotImplementedError("We still need to get to this!")
 
         train_ratio, test_ratio, validation_ratio = utils.check_split_ratio(
             split_ratio)
 
-        perm = np.random.permutation(self.df.index)
-        m = len(self.df.index)
+        train_indices = []
+        test_indices = []
+        validation_indices = []
+
+        if stratified:
+            if not stratum_column:
+                stratum_column = self.label_column
+            else:
+                if stratum_column not in self.df.columns:
+                    raise ValueError("Invalid strata column name")
+
+            grps = self.df.groupby(stratum_column)
+            train_index, test_index, val_index = [], [], []
+            for grp in grps:
+                group_train, group_test, group_val = \
+                    utils.rationed_split(grp, train_ratio, test_ratio,
+                                         validation_ratio)
+
+                train_indices += group_train
+                test_indices += group_test
+                validation_indices += group_val
 
-        train_end = int(train_ratio * m)
-        train_df = self.df.loc[perm[:train_end]]
-        validation_df = None  # just to shut up linter
-        if validation_ratio:
-            validation_end = int(validation_ratio * m) + train_end
-            validation_df = self.df.loc[perm[train_end:validation_end]]
-            test_start = validation_end
         else:
-            test_start = train_end
-        test_df = self.df.loc[perm[test_start:]]
 
-        train = TabularDataset(train=train_df, label_column=self.label_column)
-        test = TabularDataset(test=test_df, label_column=self.label_column)
+            train_indices, test_indices, validation_indices = \
+                utils.rationed_split(self.df, train_ratio,
+                                     test_ratio, validation_ratio)
+
+        train = TabularDataset(train=self.df.loc[train_indices],
```

Comment:
@nirtiac would you want to also include the batch parameter in this since the user can't really pass that information here?

Reply:
mmm, that applies only to dataloader, no?

Reply:
true - my mistake